### PR TITLE
Implement decommission APIs for Models

### PR DESCRIPTION
**Pull request by azabbasi**

### CODE REVIEW

File: sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsAsyncClient.java [hunk 1]
```diff
@@ -593,8 +594,8 @@ public HttpPipeline getHttpPipeline() {
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Mono<ModelData> getModel(String modelId) {
-        return withContext(context -> getModelWithResponse(modelId, context))
-            .flatMap(response -> Mono.just(response.getValue()));
+        return getModelWithResponse(modelId)
+            .map(response -> response.getValue());
```
Comment (abhipsaMisra) [inline sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsAsyncClient.java:598]: can be replaced with method reference:
```java
.map(Response::getValue);
```

File: sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsAsyncClient.java [hunk 2]
```diff
@@ -665,8 +666,8 @@ public HttpPipeline getHttpPipeline() {
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Mono<Void> deleteModel(String modelId) {
-        return withContext(context -> deleteModelWithResponse(modelId, context))
-            .flatMap(response -> Mono.just(response.getValue()));
+        return deleteModelWithResponse(modelId)
+            .map(response -> response.getValue());
```
Comment (abhipsaMisra) [inline sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsAsyncClient.java:670]: same here as well :)

File: sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsAsyncClient.java (hunk 1 above)
Comment (azabbasi) [reply sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsAsyncClient.java:598]: you fancy

File: sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsClient.java [hunk 3]
```diff
@@ -365,6 +366,21 @@ public Void deleteModel(String modelId) {
         return digitalTwinsAsyncClient.deleteModelWithResponse(modelId, context).block();
     }
 
-    //TODO: Decommission Model APIs (waiting for Abhipsa's change to come in)
+    /**
+     * Decommissions a model.
+     * @param modelId The Id of the model to decommission.
+     * @return Void
+     */
+    public Void decommissionModel(String modelId) {
```
Comment (abhipsaMisra) [inline sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsClient.java:374]: Return type here should be the keyword `void`, not the `Void` class type.

Comment (drwill-ms) [reply sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsClient.java:373]: [The more you know.](https://stackoverflow.com/questions/10839042/what-is-the-difference-between-java-lang-void-and-void)

File: sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsClient.java [hunk 4]
```diff
@@ -365,6 +365,21 @@ public Void deleteModel(String modelId) {
         return digitalTwinsAsyncClient.deleteModelWithResponse(modelId, context).block();
     }
 
-    //TODO: Decommission Model APIs (waiting for Abhipsa's change to come in)
+    /**
+     * Decommissions a model.
+     * @param modelId The Id of the model to decommission.
+     * @return Void
```
Comment (bikamani) [inline sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsClient.java:371]: No need of this

File: sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsAsyncClient.java [hunk 5]
```diff
@@ -683,11 +684,36 @@ public HttpPipeline getHttpPipeline() {
...
     PagedFlux<String> listRelationships(String digitalTwinId, String relationshipName, Context context) {
         return new PagedFlux<>(
             () -> listRelationshipsFirstPage(digitalTwinId, relationshipName, context),
             nextLink -> listRelationshipsNextPage(nextLink, context));
     }
 
+    /**
+     * Decommissions a model.
+     * @param modelId The Id of the model to decommission.
+     * @return an empty Mono
```
Comment (bikamani) [inline sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsAsyncClient.java:696]: why we need to write this return?

Comment (azabbasi) [reply sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsAsyncClient.java:696]: java doc analyzer will complain if there is no return section in the comments for publicly facing APIs

File: sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsClient.java (hunk 4 above)
Comment (timtay-microsoft) [reply sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsClient.java:371]: Yeah, javadocs only have the @ return if it returns something other than void. We'll still need them for cases like Mono<Void>, but here you can remove it

**Pull request merged**